app/geo/gui/src/ts/twxplore/gui/geo/components/map/Map.tsx

```diff
@@ -185,54 +209,88 @@ const MapImpl: React.FunctionComponent = () => {
 
       */
       case MapFeatureState.RENDERED: {
-        //loop through each feature type
-        for (const featureType of Object.values(FeatureType)) {
-          /*Check if filters need to be added for this FeatureType AND
-          We don't want to addFilters when some featuresByType lists are
-          still dirty because then the filters will be removed by removeDataset() in the LOADED case*/
-          if (
-            state.featuresByType[featureType].featureTypeState ===
-              MapFeatureTypeState.NEEDS_FILTERS &&
-            !hasDirtyFeatures
-          ) {
+        /*We don't want to work with rendered features when some featuresByType lists are
+        still dirty because theres a chance that some filters and layers will be removed by removeDataset() in the LOADED case*/
+        if (!hasDirtyFeatures)
+          //loop through each feature type
+          for (const featureType of Object.values(FeatureType)) {
             const featureTypeStrategy = getFeatureTypeStrategyByName(
               featureType
             );
-            //Dispatch the addFilter action 5 times (1 for each of frequency, timeStamp, transmissionPower, label, and locality)
-            for (var x = 0; x < 5; ++x) {
-              /*
+
+            let featureTypeStateOfFeatureType =
+              state.featuresByType[featureType].featureTypeState;
+            switch (featureTypeStateOfFeatureType) {
+              //Check if filters need to be added for this FeatureType
+              case MapFeatureTypeState.NEEDS_FILTERS: {
+                //Dispatch the addFilter action 5 times (1 for each of frequency, timeStamp, transmissionPower, label, and locality)
+                for (var x = 0; x < 5; ++x) {
```

Comment:
This is very fragile. It should be a count somewhere, maybe relative the available attribute strategies?